storage/google/cloud/storage/bucket.py

```diff
@@ -96,6 +62,37 @@ def __init__(self, bucket, page_token=None, max_results=None,
             page_token=page_token, max_results=max_results,
             extra_params=extra_params)
 
+    def _item_to_value(self, item):
+        """Convert a JSON blob to the native object.
+
+        :type item: dict
+        :param item: An item to be converted to a blob.
+
+        :rtype: :class:`.Blob`
+        :returns: The next blob in the page.
+        """
+        name = item.get('name')
+        blob = Blob(name, bucket=self.bucket)
+        blob._set_properties(item)
+        return blob
+
+    def _update_page(self):
```

Comment:
@jonparrott Would also like some advice input here. Taking `_item_to_value` away from the `Page` class wasn't a big deal, we could just move it over. But, taking away `Page.__init__` (for subclasses) made it so that custom behavior (like below) could not exist.

I'm not sure how I feel about asking subclasses to implement both `_item_to_value` and `_update_page` in these cases.